dcicutils/deployment_utils.py

```diff
@@ -18,20 +18,45 @@ def main():
 
 """
 
+import argparse
 import datetime
 import glob
 import io
 import json
+import logging
 import os
 import pkg_resources
 import re
 import subprocess
 import sys
 import toml
-import argparse
 
-from dcicutils.env_utils import get_standard_mirror_env, data_set_for_env, get_bucket_env, INDEXER_ENVS
-from dcicutils.misc_utils import PRINT
+from .env_utils import get_standard_mirror_env, data_set_for_env, get_bucket_env, INDEXER_ENVS
+from .misc_utils import PRINT
+
+
+def boolean_setting(settings, key, default=None):
+    """
+    Given a setting from a .ini file and returns a boolean interpretation of it.
+    - Treats an actual python boolean as itself
+    - Using case-insensitive comparison, treats the string 'true' as True and both 'false' and '' as False.
+    - If an element is missing,
+    - Treats None the same as the option being missing, so returns the given default.
+    - Raises an error for any non-standard value. This is harsh, but it should not happen.
+    """
+    if key not in settings:
+        return default
+    setting = settings[key]
+    if isinstance(setting, str):
+        setting_lower = setting.lower()
+        if setting_lower in ("", "false"):
+            return False
+        elif setting_lower == "true":
+            return True
+        else:
+            return setting
```

Comment:
Should we not coerce whatever this could be to False?

Reply:
Again, it already functions as a false value. I really wanted to raise an error here, but was nervous about where that error would be perceived.  Returning a stray value might make it easier to spot the error downstream than if we coerce it.

Reply:
Again seems reasonable.